src/ui/CMakeLists.txt

```diff
@@ -162,10 +164,22 @@ else() # native build
             yaml-cpp::yaml-cpp
             core
             client
+            services
             app_header
             utils
             ui_assets
             sample_dashboards
             fonts)
   target_compile_definitions(${target_name} PRIVATE BLOCKS_DIR="${GNURADIO_PREFIX}/share/gnuradio/grc/blocks/")
 endif()
+
+target_link_libraries(${target_name} PRIVATE function2::function2)
+target_include_directories(${target_name} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/../service/acquisition) # for daq_api.hpp
+
+add_executable(remotesignalsources_test remotesignalsources.cpp)
+target_compile_options(remotesignalsources_test PUBLIC -DCOMPILEWITHMAIN=1)
+target_link_libraries(
+  remotesignalsources_test
+  PRIVATE client
+          services
+          imgui)
```

Comment:
We usually use the `*_test` targets for unittests that can be run in the CI, this is more like an example. It also seems like it didn't survive the latest refactorings, and fails for me with

```
/home/akrimm/workspace-cpp/opendigitizer/src/ui/cmake-build-debug-native/remotesignalsources_test
RestClient::request(http://localhost:8080/dns?signal_rate=nan&signal_type=&signal_unit=&signal_name=&service_type=&service_name=&port=-1&hostname=&protocol=)

remotesignalsources_test: /home/akrimm/workspace-cpp/opendigitizer/src/ui/cmake-build-debug-native/_deps/imgui-src/imgui_tables.cpp:1413: void ImGui::TableSetupColumn(const char*, ImGuiTableColumnFlags, float, ImGuiID): Assertion `table != __null && "Need to call TableSetupColumn() after BeginTable()!"' failed.
```

I see that this was probably useful/needed for development and also might be in the future, but at the same time it seems difficult to keep up to date. It should probably be removed along with the default disabled main function in remotesignalsources.